lib/crypto/client.cpp

```diff
@@ -378,6 +378,126 @@ OlmClient::create_outbound_session(const std::string &identity_key, const std::s
         return session;
 }
 
+std::unique_ptr<SAS>
+OlmClient::sas_init()
+{
+        return std::make_unique<SAS>();
+}
+
+SAS::SAS()
+{
+        this->sas       = create_olm_object<SASObject>();
+        auto random_buf = create_buffer(olm_create_sas_random_length(sas.get()));
+
+        const int ret = olm_create_sas(this->sas.get(), random_buf.data(), random_buf.size());
+
+        if (ret == -1)
+                throw olm_exception("create_sas_instance", this->sas.get());
+}
+
+std::string
+SAS::public_key()
+{
+        auto pub_key_buffer = create_buffer(olm_sas_pubkey_length(this->sas.get()));
+
+        const int ret =
+          olm_sas_get_pubkey(this->sas.get(), pub_key_buffer.data(), pub_key_buffer.size());
+
+        if (ret == -1)
+                throw olm_exception("get_public_key", this->sas.get());
+
+        return to_string(pub_key_buffer);
+}
+
+void
+SAS::set_their_key(std::string their_public_key)
+{
+        auto pub_key_buffer = to_binary_buf(their_public_key);
+
+        const int ret =
+          olm_sas_set_their_key(this->sas.get(), pub_key_buffer.data(), pub_key_buffer.size());
+
+        if (ret == -1)
+                throw olm_exception("get_public_key", this->sas.get());
+}
+
+std::vector<int>
+SAS::generate_bytes_decimal(std::string info)
+{
+        auto input_info_buffer = to_binary_buf(info);
+        auto output_buffer     = create_buffer(5);
```

Comment:
iirc, create_buffer creates a buffer filled with random numbers. You may want to call BinaryBuf(5) instead, I think, if you are not creating a cryptographic key but are just using the buffer as a resource.